lib/matplotlib/patches.py

```diff
@@ -1113,6 +1118,11 @@ def __init__(self, x, y, dx, dy, width=0.001, length_includes_head=False,
             head_length = 1.5 * head_width
 
         distance = np.sqrt(dx ** 2 + dy ** 2)
+
+        # Account for divide by zero
+        if distance == 0:
```

Comment:
This feels wrong to me, shouldn't be we special casing it where we divide by it?